src/devices/secplus_v1.c

```diff
@@ -0,0 +1,432 @@
+
+Binary | Trinary
+--- | ---
+`0 0 0 0` | invalid
+`0 0 0 1` | 0
+`0 0 1 1` | 1
+`0 1 1 1` | 2
+`1 1 1 1` | invalid
+
+000100110111011100110001 -> 0001 0011 0111 0111 0011 0001 -> 1 11 111 111 11 1 -> [0, 1,2, 2, 1, 0]
+
+The pattens `1 1 1 1` or `0 0 0 0` should never happen
+
+*/
+
+static int _decode_v1_half(uint8_t *bits, uint8_t *result, int verbose)
+{
+    uint8_t *r;
+    int x = 0;
+
+    r = result;
+
+    for (int i = 0; i < 11; i++) {
+        // fprintf(stderr, "\nbin X = {%ld} %s\n", strlen(binstr), binstr);
+        for (int j = 0; j < 8; j++) {
+            int k = (bits[i] << j) & 0x80;
+            // fprintf(stderr, "k == %d\n", k);
+            if (k) {
+                x++;
+            }
+            else {
+                if (x == 0) {
+                    continue;
+                }
+                else if (x == 1) {
+                    *r++ = 0;
+                    // fprintf(stderr, "\nbin 0 = {%ld} %s\n", strlen(binstr), binstr);
+                }
+                else if (x == 2) {
+                    *r++ = 1;
+                    // fprintf(stderr, "\nbin 1 = {%ld} %s\n", strlen(binstr), binstr);
+                }
+                else if (x == 3) {
+                    *r++ = 2;
+                    // fprintf(stderr, "\nbin 2 = {%ld} %s\n", strlen(binstr), binstr);
+                }
+                else { // x > 3
+                    if (verbose)
+                        fprintf(stderr, "Error x == %d\n", x);
+                    return -1; // DECODE_FAIL_SANITY
+                }
+                x = 0;
+            }
+        }
+    }
+
+    return (int)result[0];
+}
+
+static const uint8_t preamble_1[1] = {0x02};
+static const uint8_t preamble_2[1] = {0x07};
+
+/** @fn static int find_next(bitbuffer_t *bitbuffer, int cur_index)
+
+    find index of next bursts/packets in bitbuffer
+
+    The transmissions do not have a magic number or preamble.
+
+    They all start with a '0' or a '2'  represented at 0001. and 0111.
+    since all nibbles start with 0 we can look for bytes
+    000 + 0001 + 0 and 000 + 0111 + 0 for the start of a transmission
+    (or just the 0001 and 0111 at the start of a bitbuffer)
+*/
+
+static int find_next(bitbuffer_t *bitbuffer, int cur_index)
+{
+
+    // int search_index;
+    int search_index_1;
+    int search_index_2;
+
+    if (cur_index == 0 && ((bitbuffer->bb[0][0] & 0xf0) == 0x10 || (bitbuffer->bb[0][0] & 0xf0) == 0x70))
+        return 0;
+
+    if (cur_index == 0 && ((bitbuffer->bb[0][0] & 0xE0) == 0xe0 || (bitbuffer->bb[0][0] & 0xc0) == 0x80))
+        return 0;
+
+    search_index_1 = bitbuffer_search(bitbuffer, 0, cur_index, preamble_1, 8);
+    search_index_1 += 3;
+
+    search_index_2 = bitbuffer_search(bitbuffer, 0, cur_index, preamble_2, 8);
+    search_index_2 += 3;
+
+    // return first match in buffer
+    return (search_index_1 < search_index_2 ? search_index_1 : search_index_2);
+}
+
+// max age for cache in us
+#define CACHE_MAX_AGE 75000
```

Comment:
75ms is too short; the packets are actually spaced 100ms apart, so the only reason this currently works is that rtl_433 defaults to receiving samples from the RTL-SDR 524ms at a time. If the receive buffer in shortened (e.g. `-b 16384) then no data is received because the cache is always considered stale.

800000 might be a good value here, since the whole transmission lasts 800ms after a button press, and that value also happens to be larger than rtl_433's default RTL-SDR buffer size.

Reply:
ok 800000 it is

Reply:
done